src/mixins/filing-template-mixin.ts

```diff
@@ -1506,7 +1518,12 @@ export default class FilingTemplateMixin extends DateMixin {
         if (!party.mailingAddress.streetAddressAdditional) delete party.mailingAddress.streetAddressAdditional
         if (!party.mailingAddress.deliveryInstructions) delete party.mailingAddress.deliveryInstructions
       }
-
+      // If the partyType is organization, remove firstName, middleName and lastName
+      if (party.officer.partyType === PartyTypes.ORGANIZATION) {
+        delete party.officer.firstName
+        delete party.officer.lastName
+        delete party.officer.middleName
+      }
       // NB: at this time, do not convert party from "middleName" to "middleInitial"
 
```

Comment:
I saw this note, not sure whether it matters?
The UI has "middleName", while the schema has "middleInitial".
In addition, in the person schema, it doesn't have taxID and Identifier, while we have them in the filing JSON

Reply:
I believe there is still a mismatch between what the UI calls "middle name" and what the BE calls it. In fact, I think one of the API endpoints returns one property and another endpoint calls it something else. In some places in the UI, we handle this...

That looks like a comment I wrote but I don't remember why. (Note to self: say why when you write something like this.)

I'm guessing this conversion is not needed because the API accepts middleName.

The schema doesn't prevent extra properties.

Reply:
I think you should just leave this comment here for now.

Please separate the logical code blocks with a blank line (before 1521 and after 1526).

Reply:
> I saw this note, not sure whether it matters? The UI has "middleName", while the schema has "middleInitial".

I filed a Limited Restoration Extension with a new applicant with a middle name and the output looks correct, so I assume the BE code is working correctly as is. No need to change anything.

![image](https://github.com/user-attachments/assets/60a00961-2fa1-40e9-bcd8-ea32d0c5f6aa)

Reply:
Resolving this conversation.